### ISSUE TITLE
Line 114 in train_AEI.py

### ISSUE THREAD

**Issue by ocastan** [taotaonice/FaceShifter#13]
Should you discard gradient calculation like in line 97 when computing arcface embeddings ?

**Comment by taotaonice** [comment]
Yes. We would not like to update parameters of arcface.

**Comment by ocastan** [comment]
According to my experience that was a bad idea: I've been trying multiple runs while appending detach() to Y_aligned and applying torch.no_grad() to the second call to arcface; it prevented L_id convergence and so face swap.
